packages/deployment-server-ui/src/pages/Devices.tsx

```diff
@@ -1,9 +1,90 @@
+import { Device, DeviceType } from ".prisma/client";
+import { Table } from "antd";
+import { useEffect, useState } from "react";
 import defineFunctionalComponent from "../util/defineFunctionalComponent";
 
 export default defineFunctionalComponent(function Devices() {
+  const [devices, setDevices] = useState<Device[]>(
+    Array(15).fill({})
+  );
+  const [deviceTypes, setDeviceTypes] = useState<DeviceType[]>(
+    Array(15).fill({})
+  );
+
+  const [filters, setFilters] = useState<any>({});
+  
+  function handleChange (pagination: any, filters: any, sorter: any) {
+    console.log('Various parameters', pagination, filters, sorter);
+    setFilters(filters);
+    window.history.replaceState(null, "Filtered Results", "/devices?type-" + filters.deviceTypeId + "&status=" + filters.status)
```

Comment:
https://reactrouter.com/docs/en/v6/examples/search-params

Don't do this manual handling, it doesn't work right now either, see `"?type-\"`